python/ray/data/read_api.py

```diff
@@ -400,6 +400,10 @@ def read_datasource(
     # TODO(hchen/chengsu): Remove the duplicated get_read_tasks call here after
     # removing LazyBlockList code path.
     read_tasks = datasource_or_legacy_reader.get_read_tasks(requested_parallelism)
+
+    if len(read_tasks) < parallelism:
+        parallelism = len(read_tasks)
+
```

Comment:
What happens if we don't add this? I think it might be causing some tests to fail:
```

[2025-01-28T19:29:11Z] FAILED python/ray/data/tests/test_splitblocks.py::test_small_file_split - ass...
--
  | [2025-01-28T19:29:11Z] FAILED python/ray/data/tests/test_splitblocks.py::test_large_file_additional_split


```

Reply:
oh, good catch, let me try